lib/dry/system/component.rb

```diff
@@ -185,7 +178,24 @@ def namespace
 
       # @api private
       def auto_register?
-        !!options.fetch(:auto_register) { true }
+        auto_register = options[:auto_register]
```

Comment:
The tricky thing with this is that the `auto_register` option may come from in-file magic comments, which will be booleans as a result of its simple built-in coercions.

If we didn't support the magic comments, then the `Config::ComponentDir` class would be a good place to normalise these values, but because it would be the only place this value is configured.

Given this, do you think we should still try to do it? I like the outcome you're suggesting, but this arrangement means `Component` itself is really the only feasible place to normalise the value, and I feel like any work to do that here would probably wind up being worse/more verbose than what's currently in place. What do you think?

Reply:
Happy to continue this conversation and find a nice way to do this in a follow-up PR, btw! But I think I might go ahead and merge this so we can begin our testing from `master` sooner rather than later :)

Reply:
@flash-gordon Actually, here's what I've done just to tidy up the behaviour before considering further options:

```ruby
      # @api private
      def auto_register?
        callable_option?(options[:auto_register])
      end

      # @api private
      def memoize?
        callable_option?(options[:memoize])
      end

      private

      def callable_option?(value)
        if value.respond_to?(:call)
          !!value.call(self)
        else
          !!value
        end
      end
```